src/encoder/init.cpp

```diff
@@ -31,7 +31,12 @@ NAN_METHOD(encodeToPngBuffer) {
     int compression = args[3].As<Integer>()->Value();
     bool interlaced = args[4]->BooleanValue();
     bool trans = args[5]->BooleanValue();
-    NanCallback * callback = new NanCallback(args[6].As<Function>());
+
+    int metadata_len = args[6].As<String>()->Utf8Length();
```

Comment:
Good catch! Looks like it actually doesn't include the null termination character, so I account for that in the new changes.